src/cdk/table/coalesced-style-scheduler.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, NgZone} from '@angular/core';
+import {from, Observable} from 'rxjs';
+
+@Injectable()
+export class CoalescedStyleScheduler {
```

Comment:
IIUIC, this is essentially a step towards a more generic way of scheduling and coordinating DOM updates in a way that prevents layout thrashing, but it is missing some core functionality from that. This has already been implemented in other libraries such as Google Closure - https://github.com/google/closure-library/blob/master/closure/goog/dom/animationframe/animationframe.js. Perhaps we can port that for a more flexible/powerful tool?

Reply:
If Jeremy and co are up for it, I can move this to a more common part of the cdk and have it be providedIn: 'any' rather than scoped to a table. For the purposes of this PR, I'm not looking to turn it into a fully-featured library, though.

Reply:
I'd rather keep it more tightly scoped for the time being. I believe this is something we should tackle at the framework level rather than at the library level, since then everyone can get the benefits without having to jump through extra hoops.